[12.0] [MIG] Account Check Deposit

ThomasBinsfeld opened the pull request:
Migration of account_check_deposit from 11.0 to 12.0.

bguillot committed:
[INIT] init module to manage check deposits

sebastienbeau committed:
[FIX] fix remove pdb call

chafique-delli committed:
[IMP]account_check_deposit: add the print option webkit

bguillot committed:
[FIX+IMP] account_check_deposit: Several things:

* Clean code
* Add access rules for account_check_deposit
* Change print button name

sebastienbeau committed:
[FIX+REF+IMP] account_check_deposit:

* [FIX] fix datamodel structure, indeed check_payment_ids should be a m2o. In the view we still keep the widget
* [REF] remove useless state cancel, only draft/done is sufficent for now, forbid to delete a done deposit, fix move_line name need to be the equal to the line.ref for the reconciliation
* [IMP] improve reconciliation between the check and the check deposit
* [FIX] set correctly the date on the move line generated
* [FIX] fix reconciliation of check
* [IMP] check deposit improve view and add tab for move line generated
* [IMP] add total amount and flag to know if the deposit have been reconcile, the best will be to add a workflow
* [FIX] fix error when replaying commit
* [REF] remove dependency on sale_quick_payment as check can be registred in various way like native voucher
* [CLEAN] pep8 cleanup no code change
* [REF] clean remove useless code, default value are called during the create
* [FIX] fix wrong spelling
* [REF] V7 clean up

Commit:
Add support for multi-currency
Works with a single account journal
Usability and many other enhancements

alexis-via committed:
Port to v8
Convert webkit report to qweb

sebastienbeau committed:
[FIX] fix domain, give the posibility to use a different journal (one for the payment, one for the deposit), hide unreconcile entry which a link to a deposit

rschnapka committed:
german translations upto invoice currency

legalsylvain committed:
[FIX] move translation file in unported folders

sebastienbeau committed:
[PORT] start porting account_check_deposit

oca-transbot committed:
OCA Transbot updated translations from Transifex

alexis-via committed:
[REF] account_check_deposit : Port to new API

* Move description from __openerp__.py to README.rst
* Move files in sub-directories
* FIX access rights issue when user was not accounting manager

oca-transbot committed:
OCA Transbot updated translations from Transifex

mourad-ehm committed:
[MIG] account_check_deposit: Migration to 9.0

* create move and move line at the same time
* add tests

florian-dacosta committed:
Fix payment field + automatic reconciliation on validation

alexis-via committed:
[MIG] account_check_deposit: Migration to v10

* Add option to have the counter-part of the deposit move directly to the bank account
  Use account.config.settings page
* Remove data that create a bank journal because it blocks the loading of the chart of accounts when
  the module is installed at the same time as the chart of account (via dependencies)

oca-transbot committed:
OCA Transbot updated translations from Transifex

alexis-via committed:
[FIX+imp] account_check_deposit: no auto-post move + use the bank journal if the offsetting account is the bank account

oca-transbot committed:
OCA Transbot updated translations from Transifex

pedrobaeza committed:
[FIX] account_check_deposit: Change company currency in tests by SQL

For avoiding constraint.

oca-travis committed:
[UPD] Update account_check_deposit.pot

pedrobaeza committed:
[MIG] account_check_deposit: Migration to 11.0

oca-travis committed:
[UPD] Update account_check_deposit.pot

OCA-git-bot committed:
[UPD] README.rst

rschnapka committed:
Translated using Weblate (German)

Currently translated at 100,0% (61 of 61 strings)

Translation: account-financial-tools-11.0/account-financial-tools-11.0-account_check_deposit
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-11-0/account-financial-tools-11-0-account_check_deposit/de/

oca-transbot committed:
Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

ThomasBinsfeld committed:
[MIG] Account Check Deposit to 12.0

ThomasBinsfeld committed:
[REF] Gitignore: .eggs

sbidoul reviewed (changes requested):
The report is broken: `ValueError: External ID not found in the system: report.internal_layout`

nikul-serpentcs commented on account_check_deposit/__manifest__.py:
```diff
@@ -0,0 +1,34 @@
+        'views/account_deposit_view.xml',
+        'views/account_move_line_view.xml',
+        'views/res_config_settings_views.xml',
+        'security/ir.model.access.csv',
```
@ThomasBinsfeld `security` files should be first

ThomasBinsfeld replied:
Done

nikul-serpentcs reviewed (commented):
Minor change

ThomasBinsfeld commented:
@sbidoul Fixed

sbidoul commented:
Tested on runbot, works fine.

alexis-via committed:
account_check_deposit: Don't force 'name' of account.move

Add print button (feature up-ported from v10)
Add option to post moves (feature up-ported from v10)
Sequence now takes into account the deposit date
Customize the filename of the report
Update README with v12 labels

alexis-via commented:
I improved this PR by up-porting from v10 1 important fix and several new features, cf my PR on acsone's branch https://github.com/acsone/account-financial-tools/pull/7
Once my PR is merged, this PR will be ready to merge. Code review + tests OK.

nikul-serpentcs commented on account_check_deposit/models/account_deposit.py:
```diff
@@ -0,0 +1,263 @@
+        required=True, states={'done': [('readonly', '=', True)]})
+    journal_default_account_id = fields.Many2one(
+        'account.account', related='journal_id.default_debit_account_id',
+        string='Default Debit Account of the Journal', readonly=True)
```
@ThomasBinsfeld no need `readonly=True`

nikul-serpentcs commented on account_check_deposit/models/account_deposit.py:
```diff
@@ -0,0 +1,263 @@
+        states={'done': [('readonly', '=', True)]})
+    line_ids = fields.One2many(
+        'account.move.line', related='move_id.line_ids',
+        string='Lines', readonly=True)
```
same here

nikul-serpentcs commented on account_check_deposit/models/account_deposit.py:
```diff
@@ -0,0 +1,263 @@
+        states={'done': [('readonly', '=', True)]},
+        default=lambda self: self.env['res.company']._company_default_get(
+            'account.check.deposit'))
+    total_amount = fields.Float(
```
same here or others `compute`

ThomasBinsfeld replied:
@nikul-serpentcs Thanks for your review. And won't add this kind of refactoring in this migration pull request since it's just cosmetic changes. Feel free to open a refactoring PR after the merge.

nikul-serpentcs reviewed (changes requested):
Improve Code

gregory-moka commented:
Hello,
no way to access account_check_deposit on runbut (no menus, custom fields blocked)

ThomasBinsfeld commented:
@gregory-moka Hello, did you check `Show Full Accounting Features` on you user?

gregory-moka commented:
@ThomasBinsfeld Ok, it wasn't set as default on the runbot session. Thanks
